scripts/cmake/z_vcpkg_function_arguments.cmake

```
@@ -0,0 +1,47 @@
+#[===[.md:
+# z_vcpkg_function_arguments
+
+Get a list of the arguments which were passed in.
+Unlike `ARGV`, which is simply the arguments joined with `;`,
+so that `(A B)` is not distinguishable from `("A;B")`,
+this macro gives `"A;B"` for the first argument list,
+and `"A\;B"` for the second.
+
+```cmake
+z_vcpkg_function_arguments(<out-var> [<N>])
+```
+
+`z_vcpkg_function_arguments` gets the arguments between `ARGV<N>` and the last argument.
+`<N>` defaults to `0`, so that all arguments are taken.
+
+## Example:
+```cmake
+function(foo_replacement)
+    z_vcpkg_function_arguments(ARGS)
+    foo(${ARGS})
+    ...
+endfunction()
+```
+#]===]
+macro(z_vcpkg_function_arguments OUT_VAR)
+    if("${ARGC}" EQUAL 1)
+        set(z_vcpkg_function_arguments_FIRST_ARG 0)
+    elseif("${ARGC}" EQUAL 2)
+        set(z_vcpkg_function_arguments_FIRST_ARG "${ARGV1}")
```

```suggestion
    if(${ARGC} EQUAL 1)
        set(z_vcpkg_function_arguments_FIRST_ARG 0)
    elseif(${ARGC} EQUAL 2)
        set(z_vcpkg_function_arguments_FIRST_ARG "${ARGV1}")
```

do you need the quotes here ?

---

Technically, no, but I don't want to have even a whiff of double-expansion.

Side note, the reason that this doesn't use `if (ARGC EQUAL 1)` is that that will use the ARGC of the containing function.

---

> Side note, the reason that this doesn't use if (ARGC EQUAL 1) is that that will use the ARGC of the containing function.

Yeah according to the macro docs `ARGC` is not a variable in a macro. CMake is basically doing string replacement with anything which is `${ARGC}` within a macro.

---

Yeah, exactly :)